Create a release-notes PR template to be used by the release tracking ticket (#693)
stackabletech/documentation: .github/PULL_REQUEST_TEMPLATE/release-notes.md

---
name: Release Notes
about: This template can be used to track the progress of the SDP Release Notes compilation

> title: "chore(tracking): Release Notes for SDP XX.(X)X"

Reviewer: suggestion: We should start using a semantic version here, eg:

```suggestion
title: "chore(tracking): Release Notes for SDP YY.M.X"
```

This indicates the use of two-digit years, one/two-digit months (without leading zero), and finally the patch release number.

Reviewer: It might be worth to unify this across all issue and PR templates, as well as labels, changelogs, etc.

assignees: ''
---

<!--
    DO NOT REMOVE THIS COMMENT. It is intended for people who might copy/paste from the previous release issue.
    This was created by an issue template: https://github.com/stackabletech/issues/issues/new/choose.
-->

> ## Release Notes for SDP XX.(X)X

Reviewer: suggestion: We should add a Markdown comment here stating that `XX.(X)X` and `YY(YY)-MM` should be replaced.

suggestion: We should also be consistent with which placeholder we use `XX` vs `YY-MM`.

Author: Yeah, I was also thinking about consistency. We should go with `Y`, `M`, and `X` as placeholders (repeated as needed, and using `(` and `)` as needed)

Author: Do you mean like:

```suggestion

<!-- Release placeholders YY.(M)M, YY.(M)M.X, YY(YY)-MM should be replaced. -->
## Release Notes for SDP YY.(M)M
```

Reviewer: I'm fine with the extra `(M)`, but we should also add `.X` at the end.

Author: As discussed, let's drop the parentheses.

Author: Done in 5c081cf

```[tasklist]
#### Release note compilation tasks
- [ ] Check [Issues](https://github.com/search?q=org%3Astackabletech+label%3Arelease-note%2Crelease-note%2Faction-required+label%3Arelease%2FYYYY-MM%2Cscheduled-for%2FYYYY-MM&type=issues&ref=advsearch) for Product and Platform release notes
- [ ] Check [PRs](https://github.com/search?q=org%3Astackabletech+label%3Arelease-note%2Crelease-note%2Faction-required+label%3Arelease%2FYYYY-MM%2Cscheduled-for%2FYYYY-MM&type=pullrequests&ref=advsearch) for Product and Platform release notes
- [ ] Optionally check the [Changelogs](https://github.com/search?q=org%3Astackabletech+path%3A*CHANGELOG.md+%22YY.MM.X%22&type=code&ref=advsearch) in case release notes were missed
- [ ] Compile list of new product versions that are supported and compile a list of new product features to include in the Release Highlights
- [ ] Upgrade guide: Document how to use stackablectl to uninstall all and install new release
- [ ] Upgrade guide: Document how to use helm to uninstall all and install new release
- [ ] Upgrade guide: Every breaking change of all our operators

> - [ ] Upgrade guide: List dropped supported product versions (if there are some)

Reviewer: suggestion: `dropped supported` sounds a little confusing in my opinion. Use something like this instead:

```suggestion
- [ ] Upgrade guide: List removed product versions (if there are some)
```

> - [ ] Upgrade guide: List dropped supported operators (if there are some)

Reviewer: suggestion: Similar to above.

```suggestion
- [ ] Upgrade guide: List removed operators (if there are some)
```

> - [ ] Upgrade guide: List supported k8s versions

Reviewer: suggestion: Either use `K8s` or `Kubernetes`

```suggestion
- [ ] Upgrade guide: List supported Kubernetes versions
```

```

Once the above items are complete, the following checks need to be done

> Each task focuses on a specific element

Author: Needs revising

Author: ```suggestion
Each of the following tasks focuses on a specific goal and should be done once the items above have been completed.
Once the above items are complete, the following checks should be done.
```

Author: Done in 0c2021d

---
name: Release Notes
about: This template can be used to track the progress of the SDP Release Notes compilation
title: "chore(tracking): Release Notes for SDP YY.M.X"
assignees: ''
---

<!--
    DO NOT REMOVE THIS COMMENT. It is intended for people who might copy/paste from the previous release issue.
    This was created by an issue template: https://github.com/stackabletech/issues/issues/new/choose.
-->

> [!CAUTION]
> Please assign the applicable `scheduled-for/YY.M.X` label.

<!-- Release placeholders YY.M.X should be replaced. -->
## Release Notes for SDP YY.M.X

> [!TIP]
> - Use the commented out template headings in [release-notes][template].
> - Begin each sentence on a new line. This helps with review suggestions and diffing.
> - Use xrefs for links to other parts of the documentation so that they remain valid across versions.

[template]: https://github.com/stackabletech/documentation/blob/8dc93f28ac6d20a587f54d0a697c71fe47e8643a/modules/ROOT/pages/release-notes.adoc?plain=1#L11-L56

```[tasklist]
#### Release note compilation tasks

> - [ ] Check [Issues](https://github.com/search?q=org%3Astackabletech+label%3Arelease-note%2Crelease-note%2Faction-required+label%3Arelease%YY.M.X%2Cscheduled-for%YY.M.X&type=issues&ref=advsearch) for Product and Platform release notes

Reviewer: question: I don't think we need `&ref=advsearch` here (and links below) right?

Author: Not sure, I just chose the one Github generated to play it safe.

Want it removed?

Reviewer: I think this basically just says "This search term came from the advanced search form". So yeah I would say we can remove it.

Author: Fixing it here with the other two lines: https://github.com/stackabletech/documentation/pull/693/files#r1879673465

- [ ] Check [PRs](https://github.com/search?q=org%3Astackabletech+label%3Arelease-note%2Crelease-note%2Faction-required+label%3Arelease%YY.M.X%2Cscheduled-for%YY.M.X&type=pullrequests&ref=advsearch) for Product and Platform release notes

> - [ ] Optionally check the [Changelogs](https://github.com/search?q=org%3Astackabletech+path%3A*CHANGELOG.md+%22YY.MM.X%22&type=code&ref=advsearch) in case release notes were missed

Reviewer: suggestion:

```suggestion
- [ ] Optionally check the [Changelogs](https://github.com/search?q=org%3Astackabletech+path%3A*CHANGELOG.md+%22YY.M.X%22&type=code&ref=advsearch) in case release notes were missed
```

Author: Fixing it here: https://github.com/stackabletech/documentation/pull/693/files#r1879673465

Author: Fixing this here so it can go in one change: https://github.com/stackabletech/documentation/pull/693/files#r1879673465
